sycl/source/detail/ur_device_info_ret_types.inc

```diff
@@ -0,0 +1,189 @@
+MAP(UR_DEVICE_INFO_HOST_PIPE_READ_WRITE_SUPPORT, ur_bool_t)
+MAP(UR_DEVICE_INFO_MAX_REGISTERS_PER_WORK_GROUP, uint32_t)
+MAP(UR_DEVICE_INFO_IP_VERSION, uint32_t)
+MAP(UR_DEVICE_INFO_VIRTUAL_MEMORY_SUPPORT, ur_bool_t)
+MAP(UR_DEVICE_INFO_ESIMD_SUPPORT, ur_bool_t)
+MAP(UR_DEVICE_INFO_COMPONENT_DEVICES, std::vector<ur_device_handle_t>)
+MAP(UR_DEVICE_INFO_COMPOSITE_DEVICE, ur_device_handle_t)
+MAP(UR_DEVICE_INFO_GLOBAL_VARIABLE_SUPPORT, ur_bool_t)
+MAP(UR_DEVICE_INFO_USM_POOL_SUPPORT, ur_bool_t)
+MAP(UR_DEVICE_INFO_NUM_COMPUTE_UNITS, uint32_t)
+MAP(UR_DEVICE_INFO_PROGRAM_SET_SPECIALIZATION_CONSTANTS, ur_bool_t)
+MAP(UR_DEVICE_INFO_USE_NATIVE_ASSERT, ur_bool_t)
+MAP(UR_DEVICE_INFO_CURRENT_CLOCK_THROTTLE_REASONS, ur_device_throttle_reasons_flags_t)
+MAP(UR_DEVICE_INFO_FAN_SPEED, int32_t)
+MAP(UR_DEVICE_INFO_MIN_POWER_LIMIT, int32_t)
+MAP(UR_DEVICE_INFO_MAX_POWER_LIMIT, int32_t)
+MAP(UR_DEVICE_INFO_BFLOAT16_CONVERSIONS_NATIVE, ur_bool_t)
+
+// These aren't present in the specification, extracted from ur_api.h
+// instead.
+MAP(UR_DEVICE_INFO_2D_BLOCK_ARRAY_CAPABILITIES_EXP, ur_exp_device_2d_block_array_capability_flags_t)
+MAP(UR_DEVICE_INFO_ASYNC_USM_ALLOCATIONS_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_IMAGES_1D_USM_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_IMAGES_2D_USM_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_IMAGES_GATHER_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_IMAGES_SHARED_USM_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_IMAGES_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_1D_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_1D_USM_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_2D_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_2D_USM_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_3D_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_SAMPLE_1D_USM_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_SAMPLE_2D_USM_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_BINDLESS_UNIQUE_ADDRESSING_PER_DIM_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_CLUSTER_LAUNCH_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_CUBEMAP_SEAMLESS_FILTERING_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_CUBEMAP_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_EXTERNAL_MEMORY_IMPORT_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_EXTERNAL_SEMAPHORE_IMPORT_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_IMAGE_ARRAY_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_IMAGE_PITCH_ALIGN_EXP, uint32_t)
+MAP(UR_DEVICE_INFO_MAX_IMAGE_LINEAR_HEIGHT_EXP, size_t)
+MAP(UR_DEVICE_INFO_MAX_IMAGE_LINEAR_PITCH_EXP, size_t)
+MAP(UR_DEVICE_INFO_MAX_IMAGE_LINEAR_WIDTH_EXP, size_t)
+MAP(UR_DEVICE_INFO_MIPMAP_ANISOTROPY_SUPPORT_EXP, ur_bool_t)
+MAP(UR_DEVICE_INFO_MIPMAP_LEVEL_REFERENCE_SUPPORT_EXP, ur_bool_t)
+// Previous use was `float`, ur_api_t has `uint32_t` in the docstring.
+MAP(UR_DEVICE_INFO_MIPMAP_MAX_ANISOTROPY_EXP, uint32_t)
```

Comment:
@intel/bindless-images-reviewers , this either introduces a bug or fixes one that was present before. In either case your follow-up is necessary (fix UR's documentation for the actual return type in the former case or add a test in the latter one).

Reply:
I'm not sure what bug you're referring to - we intentionally set it to `float`, from SYCL to UR. It's converted to `unsigned int` for the CUDA backend. We're still discussing whether to leave it as a `float` as that's what graphics APIs use, or change it to `unsigned int` throughout. However we don't want to make this change yet without further discussion.

Reply:
The code before this PR was doing.

```
float val;
adapter->callUR(UR_DEVICE_INFO_MIPMAP_MAX_ANISOTROPY_EXP, &val, sizeof(val))
```

with UR writing bits as `uint32_t` with SYCL RT **interpreting** them as `float`, not **converting** to float. After this PR it is **converting** `uint32_t` to `float`.

Reply:
Since the decision to use both types was intentional, I'm pretty sure I'm fixing a bug here, not introducing a new one. @ProGTX , please follow up with necessary tests that are currently missing.

I'm merging this to unblock creation of subsequent PRs in my change stack.